### PR TITLE
test(dan): fix broken dan layer integration tests

### PR DESCRIPTION
**Pull request by brianp**
Description
---

- Fix any tests marked `@broken`
- Remove old layer 2 code
- Add more specific assertions for tests
- Add a CI workflow for dan layer tests to ensure PR's don't break existing functionality

Motivation and Context
---
We can't keep letting tests break and spending lots of time backdating the changes. Ensure they work, and continue to work moving forward.

How Has This Been Tested?
---
Oh- the tests have been tested alright

### CODE REVIEW

File: .github/workflows/dan_layer_integration_tests.yml [hunk 1]
```diff
@@ -78,7 +78,7 @@ jobs:
           command: build
           args: --release --bin tari_validator_node
       - name: npm ci
-        run: cd integration_tests && npm ci && cd node_modules/wallet-grpc-client && npm ci
+        run: cd integration_tests && npm ci && cd node_modules/wallet-grpc-client && npm install
```
Comment (delta1) [inline .github/workflows/dan_layer_integration_tests.yml]: Doing this means npm might resolve new packages instead of what’s in package-lock

Comment (brianp) [reply .github/workflows/dan_layer_integration_tests.yml]: Currently the package-lock is gitignored. So the best solution is likely to commit  it.

Comment (delta1) [reply .github/workflows/dan_layer_integration_tests.yml]: Yeah I thought so but I think Mike removed them a while back? 🤷‍♂️

Comment (stringhandler) [reply .github/workflows/dan_layer_integration_tests.yml]: lol it gets deleted every now and again. It should be included

Comment (brianp) [reply .github/workflows/dan_layer_integration_tests.yml]: Added!

File: integration_tests/features/ValidatorNode.feature [hunk 2]
```diff
@@ -3,61 +3,45 @@
...
-        When I wait 30 seconds
-        When I register an NFT asset with committee of 4
-        And I mine 3 blocks
-        And I create 40 NFTs
-        And I mine 3 blocks
-
-    # Broken: needs a contract definition before publishing acceptance, however this is currently not easily done because
-    # GRPC methods need to be added and you cannot use the cli for a wallet while that wallet is already running
-    @dan @critical @broken
+    @dan @critical
```
Comment (stringhandler) [inline integration_tests/features/ValidatorNode.feature]: No need to add the `@dan` tag if it's specified at the feature level already

**Review by stringhandler — DISMISSED** [review]

**Review by stringhandler — APPROVED** [review]

**Pull request merged**